src/toil/lib/docker.py

```diff
@@ -0,0 +1,238 @@
+    :param list[str] parameters: Command line arguments to be passed to the tool.
+    :param str workDir: Directory to mount into the container via `-v`. Destination convention is /data
+    :param list[str] dockerParameters: Parameters to pass to Docker. Default parameters are `--rm`,
+            `--log-driver none`, and the mountpoint `-v work_dir:/data` where /data is the destination convention.
+             These defaults are removed if docker_parmaters is passed, so be sure to pass them if they are desired.
+    :param file outfile: Pipe output of Docker call to file handle
+    :param int defer: What action should be taken on the container upon job completion?
+           docker_call.FORGO (0) will leave the container untouched.
+           docker_call.STOP (1) will attempt to stop the container with `docker stop` (useful for
+           debugging).
+           docker_call.RM (2) will stop the container and then forcefully remove it from the system
+           using `docker rm -f`.
+           The default value is None and that shadows docker_call.FORGO, unless --rm is being passed in.
+    """
+    Docker(job, tool=tool, parameters=parameters, workDir=workDir, dockerParameters=dockerParameters,
+           outfile=outfile, checkOutput=False, defer=defer)
+
+
+def dockerCheckOutput(job, tool, parameters=None, workDir=None, dockerParameters=None, defer=None):
+    """
+    Wrapper call to Docker. See docstring for Docker()
+
+    :param toil.Job.job job: The Job instance for the calling function.
+    :param str tool: Name of the Docker image to be used (e.g. quay.io/ucsc_cgl/samtools).
+    :param list[str] parameters: Command line arguments to be passed to the tool.
+    :param str workDir: Directory to mount into the container via `-v`. Destination convention is /data
+    :param list[str] dockerParameters: Parameters to pass to Docker. Default parameters are `--rm`,
+            `--log-driver none`, and the mountpoint `-v work_dir:/data` where /data is the destination convention.
+             These defaults are removed if docker_parmaters is passed, so be sure to pass them if they are desired.
+    :param int defer: What action should be taken on the container upon job completion?
+           docker_call.FORGO (0) will leave the container untouched.
+           docker_call.STOP (1) will attempt to stop the container with `docker stop` (useful for
+           debugging).
+           docker_call.RM (2) will stop the container and then forcefully remove it from the system
+           using `docker rm -f`.
+           The default value is None and that shadows docker_call.FORGO, unless --rm is being passed in.
+    :returns: Stdout from the docker call
+    :rtype: str
+    """
+    return Docker(job, tool=tool, parameters=parameters, workDir=workDir,
+                  dockerParameters=dockerParameters, checkOutput=True, defer=defer)
+
+
+def Docker(job,
```

Comment:
Also, I feel like this method should be private. What's the rationale for having it be public?